sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -126,25 +126,66 @@ static void _add_static_lladdr(gnrc_netif_t *netif)
 #endif
 }
 
+void gnrc_ipv6_nib_iface_up(gnrc_netif_t *netif)
+{
+    assert(netif != NULL);
+    gnrc_netif_acquire(netif);
+
+    _init_iface_arsm(netif);
+    gnrc_netif_init_6ln(netif);
+
+#if IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_6LN)
+    netif->ipv6.rs_sent = 0;
+#endif  /* CONFIG_GNRC_IPV6_NIB_6LN */
+    netif->ipv6.na_sent = 0;
+    if (!(gnrc_netif_is_rtr_adv(netif)) ||
+        (gnrc_netif_is_6ln(netif) && !gnrc_netif_is_6lbr(netif))) {
+        uint32_t next_rs_time = random_uint32_range(0, NDP_MAX_RS_MS_DELAY);
+
+        _evtimer_add(netif, GNRC_IPV6_NIB_SEARCH_RTR, &netif->ipv6.search_rtr,
+                     next_rs_time);
+    }
+#if IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_ROUTER)
+    else {
+        _handle_snd_mc_ra(netif);
+    }
+#endif  /* CONFIG_GNRC_IPV6_NIB_ROUTER */
+
+    gnrc_netif_release(netif);
+}
+
+void gnrc_ipv6_nib_iface_down(gnrc_netif_t *netif)
+{
+    assert(netif != NULL);
+    DEBUG("nib: Deinitialize interface %u\n", netif->pid);
+    gnrc_netif_acquire(netif);
+
+    _deinit_iface_arsm(netif);
+
+    if (!(gnrc_netif_is_rtr_adv(netif)) ||
+        (gnrc_netif_is_6ln(netif) && !gnrc_netif_is_6lbr(netif))) {
+        _evtimer_del(&netif->ipv6.search_rtr);
+    }
+#if IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_ROUTER)
+    _evtimer_del(&netif->ipv6.snd_mc_ra);
+#endif
+
+    gnrc_netif_release(netif);
+}
+
 void gnrc_ipv6_nib_init_iface(gnrc_netif_t *netif)
 {
     assert(netif != NULL);
     DEBUG("nib: Initialize interface %u\n", netif->pid);
     gnrc_netif_acquire(netif);
 
-    _init_iface_arsm(netif);
     netif->ipv6.retrans_time = NDP_RETRANS_TIMER_MS;
 #if IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_SLAAC) || IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_6LN)
     /* TODO: set differently dependent on CONFIG_GNRC_IPV6_NIB_SLAAC if
      * alternatives exist */
     netif->ipv6.aac_mode |= GNRC_NETIF_AAC_AUTO;
 #endif  /* CONFIG_GNRC_IPV6_NIB_SLAAC || CONFIG_GNRC_IPV6_NIB_6LN */
     _init_iface_router(netif);
-    gnrc_netif_init_6ln(netif);
```

Comment:
Not sure if that works... Some of the parts later in this function might rely on the interface being a 6lo-interface.

Reply:
Yes you are right, this has to stay in `gnrc_ipv6_nib_init_iface()`